tests/test_global_storageclass.py

```diff
@@ -0,0 +1,27 @@
+from tests.helm_template_generator import render_chart
+import pytest
+
+chart_files = [
+    "charts/alertmanager/templates/alertmanager-statefulset.yaml",
+    "charts/elasticsearch/templates/master/es-master-statefulset.yaml",
+    "charts/elasticsearch/templates/data/es-data-statefulset.yaml",
+    "charts/prometheus/templates/prometheus-statefulset.yaml",
+    "charts/astronomer/templates/registry/registry-statefulset.yaml",
+]
+
+
+@pytest.mark.parametrize(
+    "supported_types, expected_output", [("-", ""), ("vishnure", "vishnure")]
+)
+def test_alertmanager_global_storageclass(supported_types, expected_output):
+    """Test globalstorageclass feature of alertmanager statefulset template"""
+    docs = render_chart(
+        values={"global": {"storageClass": supported_types}},
+        show_only=chart_files,
+    )
+    assert len(docs) == 5
+    doc = docs[0]
+    assert (
+        doc["spec"]["volumeClaimTemplates"][0]["spec"]["storageClassName"]
```

Comment:
This test is only testing the first chart, because `doc = docs[0]`. Run this through a `all(...jmespath.search` style list comprehension or something so we check all the docs.

Reply:
@danielhoherd : added some updates for the test cases that may not be perfect but runs through all files now.

Reply:
I slacked you some details about how to improve it using two different ways.